department_names.py

```diff
@@ -4,61 +4,94 @@
 import logging
 import os
 import requests
-import yaml
+import csv
 
-portal_url = os.environ.get('PORTAL_URL', "https://vulekamali.gov.za/")
+portal_url = os.environ.get("PORTAL_URL", "https://vulekamali.gov.za/")
+
+department_names = {}
 
-department_names = {
-    'national': {},
-    'provincial': {},
-}
 warned = {}
 
 
-def modify_datapackage(datapackage, parameters, stats):
-    # We're not modifying the datapackage but we execute here to execute once
-    # before processing rows.
-    year_slug = parameters['financial_year']
-    sphere = parameters['sphere']
-    listing_url_path = year_slug + '/departments.yaml'
+def get_financial_year(fin_year):
+    governments = []
+    listing_url_path = fin_year + "/departments.csv"
     listing_url = portal_url + listing_url_path
     r = requests.get(listing_url)
-    r.raise_for_status()
-    response = yaml.load(r.text)
-    for government in response[sphere]:
-        department_names[sphere][government['name']] = {}
-        for department in government['departments']:
-            department_names[sphere][government['name']][department['slug']] \
-                = department['name']
+    if r.status_code != 200 and r.status_code != 404:
+        r.raise_for_status()
+    elif r.status_code == 404:
+        logging.warning(f"Departments couldn't be found in URL ({listing_url})!")
+        return None
+
+    department_names[fin_year] = {
+        "national": {},
+        "provincial": {},
+    }
+    reader = csv.DictReader(r.text.splitlines(), delimiter=",")
+    for row in reader:
+        if row["government"] == "South Africa":
```

Comment:
Please don't deduce sphere from government - rather request the data for the correct sphere using the URL shown above

Reply:
Function takes sphere as a parameter now.

https://github.com/vulekamali/treasury-pipelines/pull/33/commits/5cc6afcee6d74c4acee30b247917db1caf05b85d